src/config/stylegan_adversarial.py

```diff
@@ -0,0 +1,129 @@
+import torch
+from loguru import logger as info_logger
+from src.disk import disk
+from pathlib import Path
+from src.logger.simple import Logger
+from src.losses.vgg import VGGLoss
+from src.data.baseline import BaselineDataset
+from src.losses.vgg import VGGLoss
+from src.utils.download import download_dataset
+from src.models.embedder import ContentResnet
+from src.models.nlayer_discriminator import NLayerDiscriminator
+from src.models.stylegan import StyleBased_Generator
+from src.training.stylegan_adversarial import StyleGanAdvTrainer
+from src.storage.simple import Storage
+from src.losses.STRFL import OCRLoss
+from src.losses.typeface_perceptual import TypefacePerceptualLoss
+from torchvision import models
+from torchvision.models.resnet import BasicBlock
+from torch.utils.data import DataLoader
+
+
+class Config:
+    def __init__(self):
+        disk.login()
+
+        device = 'cuda:0' if torch.cuda.is_available() else 'cpu'
+        info_logger.info(f'Using device: {device}')
+        style_dir = Path('data/IMGUR5K')
+        download_dataset('IMGUR5K')
+        batch_size = 16
+        train_dataloader = DataLoader(BaselineDataset(style_dir / 'train'), shuffle=True, batch_size=batch_size)
+        val_dataloader = DataLoader(BaselineDataset(style_dir / 'val'), batch_size=batch_size)
+
+        total_epochs = 500
+
+        weights_folder = 'models/Stylegan (pretrained on content)'
+        if not Path(weights_folder).exists():
+            disk.download(weights_folder, weights_folder)
+
+        model_G = StyleBased_Generator(dim_latent=512)
+        model_G.load_state_dict(torch.load(f'{weights_folder}/model'))
+        model_G.to(device)
+
+        style_embedder = models.resnet18()
+        style_embedder.fc = torch.nn.Identity()
+        style_embedder = style_embedder.to(device)
+        style_embedder.load_state_dict(torch.load(f'{weights_folder}/style_embedder'))
```

Comment:
We moved content embedder to a separate file, and as I understand it style embedder also kinda is a modified resnet, maybe we should make it a separate model too?

Reply:
But do we really want to move 2 lines to other file? I think it’s okay to leave it like it is

Reply:
We can move it to `embedder.py` file.

Reply:
I would leave it here, but if you want to move it, then we should definitely move it to  embedder.py